src/FormControl/test/FormControlSpec.js

```diff
@@ -223,35 +223,30 @@ describe('FormControl', () => {
         refError = e;
         setError(e);
       };
-      const handleUnMount = () => {
-        setShow(false);
-      };
+
+      const { email } = value;
       return (
         <>
-          <button onClick={handleUnMount} id="unmount">
-            unmount
-          </button>
           <Form
             model={model}
             onChange={handleChange}
             onCheck={handleError}
             formError={error}
             formValue={value}
           >
-            {show && <FormControl id="username" name="username" shouldResetWithUnmount />}
+            {email || <FormControl id="username" name="username" shouldResetWithUnmount />}
             <FormControl id="email" name="email" />
           </Form>
         </>
       );
     };
     const { container } = render(<Wrapper />);
-
-    fireEvent.change(container.querySelector('#username'), { target: { value: 'test' } });
-    assert.deepEqual(refValue, { username: 'test', email: '' });
+    fireEvent.change(container.querySelector('#username'), { target: { value: 'username' } });
+    assert.deepEqual(refValue, { username: 'username', email: '' });
     assert.deepEqual(refError, { username: 'The length cannot exceed 2' });
-    fireEvent.click(container.querySelector('#unmount'));
-    assert.deepEqual(refValue, { email: '' });
-    assert.deepEqual(refError, {});
+    fireEvent.change(container.querySelector('#email'), { target: { value: 'email' } });
+    assert.deepEqual(refValue, { email: 'email' });
+    assert.deepEqual(refError, { email: 'The length cannot exceed 2' });
```

Comment:
The assertion in the test case has no checks to remove value and error.

Reply:
I think this assertion checked it was `{username: 'username', email: ''}` becoming {email: 'email'}. That has checked that the username value has been removed, and the error is the same.